server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Rights.java

```diff
@@ -213,6 +217,36 @@ public MailboxACL toMailboxAcl() {
             .reduce(MailboxACL.EMPTY, union);
     }
 
+    public Optional<Boolean> mayReadItems(Username username) {
+        return containsRight(username, Right.Read);
+    }
+
+    public Optional<Boolean> mayAddItems(Username username) {
+        return containsRight(username, Right.Insert);
+    }
+
+    public Optional<Boolean> mayCreateChild(Username username) {
+        return UNSUPPORTED;
+    }
+
+    public Optional<Boolean> mayRemoveItems(Username username) {
+        return containsRight(username, Right.DeleteMessages);
+    }
+
+    public Optional<Boolean> mayRename(Username username) {
+        return containsRight(username, Right.Write);
```

Comment:
I'm not sure:
```
   w - write (set or clear flags other than \SEEN and \DELETED via
       STORE, also set them during APPEND/COPY)
   k - create mailboxes (CREATE new sub-mailboxes in any
       implementation-defined hierarchy, parent mailbox for the new
       mailbox name in RENAME)
```
So rename should be creation right on the parent.

Reply:
UNSUPORTED here too then